lib/web_push_encryption/push.ex

```diff
@@ -14,28 +14,34 @@ defmodule WebPushEncryption.Push do
     * `message` is a binary payload. It can be JSON encoded
     * `subscription` is the subscription information received from the client.
        It should have the following form: `%{keys: %{auth: AUTH, p256dh: P256DH}, endpoint: ENDPOIONT}`
-    * `auth_token` is the GCM api key matching the `gcm_sender_id` from the client `manifest.json`
+    * `auth_token` [Optional] is the GCM api key matching the `gcm_sender_id` from the client `manifest.json`.
+       It is not necessary for mozzilla endpoints.
 
   ## Return value
 
   Returns the result of `HTTPoison.post`
   """
   @spec send_web_push(message :: binary, subscription :: map, auth_token :: binary) :: {:ok, any} | {:error, atom}
-  def send_web_push(message, subscription, auth_token)
-  def send_web_push(_message, _subscription, nil) do
-    raise ArgumentError, "send_web_push requires an auth_token"
+  def send_web_push(message, subscription, auth_token \\ nil)
+  def send_web_push(_message, %{endpoint: @gcm_url <> _registrationId} = _subscription, nil) do
+    raise ArgumentError, "send_web_push requires an auth_token for gcm endpoints"
   end
   def send_web_push(message, %{endpoint: endpoint} = subscription, auth_token) do
-    endpoint = String.replace(endpoint, @gcm_url, @temp_gcm_url)
+    is_gcm = String.contains? endpoint,  @gcm_url
```

Comment:
A little nitpicky, but could we keep the style consistent and add brackets to the function call please?